### ISSUE TITLE
[BUG] Image search returns Internal Server Error

### ISSUE THREAD

**Issue by Nebukadneza** [benbusby/whoogle-search#643]
Searching for something and selecting `Images` returns `Internal Server Error`

**To Reproduce**
1. Go to main search page of your instance
2. Search for `kuih galang` (… to make you hungry — or anything else)
3. Select `Images`
5. In logs, find:
```
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.045950359Z ERROR:app:Exception on /search [GET]
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046002295Z Traceback (most recent call last):
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046014193Z   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046042223Z     response = self.full_dispatch_request()
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046051749Z   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046060337Z     rv = self.handle_user_exception(e)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046069087Z   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046077918Z     reraise(exc_type, exc_value, tb)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046085891Z   File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046094794Z     raise value
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046103605Z   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046112096Z     rv = self.dispatch_request()
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046119909Z   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046128128Z     return self.view_functions[rule.endpoint](**req.view_args)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046136189Z   File "/whoogle/app/routes.py", line 98, in decorated
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046144420Z     return f(*args, **kwargs)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046152511Z   File "/whoogle/app/routes.py", line 57, in decorated
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046160548Z     return f(*args, **kwargs)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046168279Z   File "/whoogle/app/routes.py", line 315, in search
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046176201Z     response = search_util.generate_response()
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046184269Z   File "/whoogle/app/utils/search.py", line 138, in generate_response
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046192554Z     html_soup = content_filter.view_image(html_soup)
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046217564Z   File "/whoogle/app/filter.py", line 367, in view_image
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046227182Z     results_div = soup.find_all('div', attrs={'class': "nQvrDb"})[0]
whoogle_1_1413963bc3e5 | 2022-02-05T19:09:42.046236642Z IndexError: list index out of range
```

**Deployment Method**

- [X] Docker-Compose (

**Version of Whoogle Search**
- [x] Latest build from Docker Hub (`benbusby/whoogle-search:0.7.1`)

**Desktop:**
 - OS: Debian GNU/Linux Testing
 - Browser: Firefox 96.0.3

**Comment by DUOLabs333** [comment]
I can't replicate this.

**Comment by eddik1970** [comment]
This happens to me too. But only if I mark words in a normal webpage, rightclick and whoogles for them, and then choose the image-tab in Whoogle. I does not happen if I search directly in Whoogle. 

The text in the search box also changes from normal words to something like: gAAAAABiAPPicUJqGDA9iyBo86azFusQBoh-heYaHmHt9TMGaApuXOu6gi4YZYECvj-YLS72aU8ow3lz6-h7c-JVe8vVGFP4nZrBI30PipYU8GPLVO5dYiE=

I am attaching 2 images. I Lynx1, I have rightclicked Lynx in a webpage and searched Whoogle for it.
![Lynx2](https://user-images.githubusercontent.com/32521015/152771059-67206887-5f83-465e-8348-ed89f62c8649.png)
I Lynx2 I pressed Image-search (Norwegian localization=Bilder) in Lynx1-image.
![Lynx1](https://user-images.githubusercontent.com/32521015/152771060-4f4c1e50-431b-46d7-8c43-92525814c4ed.png)

**Comment by Nebukadneza** [comment]
Ack, the method of reproduction that @eddik1970 described triggers this behavior here every time too. My initial example worked in between, and now is broken again for me. I wonder if this is something about how the upstream response to whoole looks, because it’s flaky and doesn’t fail every time …

**Comment by DUOLabs333** [comment]
Can't reproduce it -- can you build from `git`?

**Comment by Nebukadneza** [comment]
That’s a bit fiddly in my setup, but if I interpret the github-actions definition correctly, `main` commits are pushed as `:latest` docker-tag, right?

So I tried `:latest` right now, and it still happens to me. Let’s focus on searches initiated via firefox rightclick->search. I’ve noticed something else, not only the link to `Images` is formatted in the weird way that @eddik1970 showed in the screenshot, but the `Next >` link also — resulting in viewing the 2nd page not working.

I’ve also tried reverting to 0.6.0 — and can confirm that 0.6.0 still works for me.

I’m a bit at a loss how to best help ATM — but thanks for trying to reproduce ^_^.

**Comment by DUOLabs333** [comment]
A lot of these recent issues are from Docker -- this may mean it's uniquely a Docker problem, or that most people here use Docker (maybe both).

**Comment by DUOLabs333** [comment]
Can you try a virtualenv.

**Comment by DUOLabs333** [comment]
How would you check that @benbusby

**Comment by benbusby** [comment]
@DUOLabs333 just checking the home page config menu to see if that option is enabled. That menu reflects the user's session configuration, which may be different from any configuration done via environment variables.

**Comment by eddik1970** [comment]
This seems to be fixed for me now! Thank you!

**Comment by eddik1970** [comment]
I was a little quick, Images works, but if I press for next page of search results I get this: 

![image](https://user-images.githubusercontent.com/32521015/153405144-436e9121-f686-40a3-8420-95ed4daf0e61.png)

**Comment by Stolz** [comment]
Issue is still present fom me. I'm using current `benbusby/whoogle-search:latest` Docker image, AKA v0.7.1 

Steps to reproduce:

1. Visit Whoogle page
2. Search for anything
3. Press 'IMAGES' in Whoogle's nav bar

The browser shows an error page with this content:
```
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

Docker logs show
```
ERROR:app:Exception on /search [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/whoogle/app/routes.py", line 100, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 59, in decorated
    return f(*args, **kwargs)
  File "/whoogle/app/routes.py", line 317, in search
    response = search_util.generate_response()
  File "/whoogle/app/utils/search.py", line 148, in generate_response
    formatted_results = content_filter.clean(html_soup)
  File "/whoogle/app/filter.py", line 122, in clean
    self.remove_site_blocks(soup)
  File "/whoogle/app/filter.py", line 130, in remove_site_blocks
    selected = soup.body.findAll(text=re.compile(search_string))
AttributeError: 'NoneType' object has no attribute 'findAll'
```

**Comment by DUOLabs333** [comment]
@Stolz I submitted a pull request that fixed it. The fact it was intermittent in the past (I just experienced it) may indicate that it's a part of Google's new layout rollout.

**Comment by Nebukadneza** [comment]
Hi,

i just hand-patched #651 into my container, and this seems to do the trick for me! As you correctly stated, the errors have been intermittent so far, so i’m going to watch this for a while … but, just before i patched/restarted, i double-checked with 0.7.1, and the error occurred immediately.

So, this is looking good i think! i’ll report back again in 1-2 days, however ^_^.

Thanks a lot for your work!

**Comment by Nebukadneza** [comment]
Thanks for closing. I’ve tested it thoroughly for a while now, and the solution from #651 indeed does the trick.

Thank you a lot for your assistance!
